http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java

```diff
@@ -595,19 +584,20 @@ private boolean isSingle(RouteInfo<?> finalRoute, Class<?> bodyClass) {
                 final RouteMatch<?> finalRoute;
 
                 // ensure the route requirements are completely satisfied
+                final HttpRequest<?> httpRequest = requestReference.get();
                 if (!routeMatch.isExecutable()) {
                     finalRoute = requestArgumentSatisfier
-                            .fulfillArgumentRequirements(routeMatch, requestReference.get(), true);
+                            .fulfillArgumentRequirements(routeMatch, httpRequest, true);
                 } else {
                     finalRoute = routeMatch;
                 }
 
-                Object body = finalRoute.execute();
+                Object body = ServerRequestContext.with(httpRequest, (Supplier<Object>) finalRoute::execute);
```

Comment:
Did you find a case where the request wasn't present? If so please add a test

Reply:
It is not present in this example app linked to the issue https://github.com/chrisparton1991/micronaut-3-redis-session-scope-issue/blob/master/src/main/java/com/example/Application.java

It seems to be lost when Micronaut Security is involved. I tried but I cannot seem to come up an example that doesn't involve engaging Micronaut security like in the sample application.